### ISSUE TITLE
Enclosing components of a contract-to-method chain in brackets breaks instrumentation.

### ISSUE THREAD

**Issue by Astrovicis** [sc-forks/solidity-coverage#179]
While instrumenting, I get the following warning and error:

Warning: The file at ./coverageEnv/contracts/Tournament.sol was identified as a Solidity Contract, but did not parse correctly. You may ignore this warning if it is not a Solidity file, or your project does not use it
Running: truffle compile  
(this can take a few seconds)...
Error parsing /Users/user/Desktop/Ethereum Development/Projects/matryx-alpha-source/coverageEnv/contracts/Tournament.sol: ParsedContract.sol:404:157: ParserError: Expected token Comma got 'Semicolon'
 ...  ... 
     ^

I'm confused because I'm getting that warning yet not importing any contracts from node_modules.
Also, when I take a look at /Users/user/Desktop/Ethereum Development/Projects/matryx-alpha-source/coverageEnv/contracts/Tournament.sol at line 404, I see the following:

( __StatementCoverageTournament('/Users/user/Desktop/Ethereum Development/Projects/matryx-alpha-source/contracts/Tournament.sol',39);

Clearly this is meant to have an extra right paren before the semicolon, but I'm really not sure what to do about that. 

This feels like a lexer problem, but maybe its me. Either way,

halp.

**Comment by Astrovicis** [comment]
Found the issue. The line is

`(MatryxPlatform(platformAddress)).invokeTournamentClosedEvent(this, _submissionAddress);`

I got rid of the generated error by changing it to:

```
MatryxPlatform platform = MatryxPlatform(platformAddress);
platform.invokeTournamentClosedEvent(this, _submissionAddress);
```

**Comment by cgewecke** [comment]
Hi, @Astrovicis - interesting! Out of curiosity, are you also getting an error when you chain the address-based instantiation to the method without any enclosing brackets? e.g:
```
MatryxPlatform(platformAddress).invokeTournamentClosedEvent(this, _submissionAddress);
```

(Rewriting your issue title to reflect what I think the cause is....).

**Comment by cgewecke** [comment]
Believe this was fixed in `0.6.x` which uses solidity-parser-antlr. Closing for housekeeping.
